projects/project-2/readings/Project 2 - SPARQL library

```diff
@@ -0,0 +1,297 @@
+
+3. Detecting Unused Object Properties (level 7 - 2 points)
+
+Title: Unused Object Property Detection
+Constraint Description: Identifies object properties that are defined but not used in any instances.
+Severity: Warning
+
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+SELECT ?property
+WHERE {
+  ?property a owl:ObjectProperty .
+  FILTER NOT EXISTS {
+    ?subject ?property ?object .
+  }
+}
+
+​The aim of this query is to identify object properties within the ontology that are defined but not utilized in any instances.​ 
+Detecting such unused properties helps maintain a clean and efficient ontology by identifying potentially redundant definitions.
+
+4. Finding Classes with Conflicting Disjointness and Subclass Relationships
+
+Title: Disjoint Subclass Conflict
+Constraint Description: Identifies classes that are marked as disjoint but share a common superclass.
+Severity: Error
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?class1 ?class2
+WHERE {
+  ?class1 owl:disjointWith ?class2 .
+  ?class1 rdfs:subClassOf ?commonSuperclass .
+  ?class2 rdfs:subClassOf ?commonSuperclass .
+}
+
+The goal of this query is to identify classes that are marked as disjoint but are simultaneously defined as subclasses of a common superclass.​ 
+This situation presents a logical inconsistency within the ontology, as disjoint classes should not share any common subclasses.
+
+5. Identifying Classes with Different Labeling Standards
+
+Title: Inconsistent Class Labeling
+Constraint Description: Detects classes with different labeling standards.
+Severity: Warning
+
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+SELECT ?class ?label1 ?label2
+WHERE {
+  ?class a owl:Class .
+  ?class rdfs:label ?label1 .
+  ?otherClass a owl:Class .
+  ?otherClass rdfs:label ?label2 .
+  FILTER(?class != ?otherClass) 
+  FILTER(?label1 != ?label2)
+}
+
+The goal of this query is to find classes within an ontology that are associated with different labeling standards.​ 
+This can be important for identifying inconsistencies in how classes are labeled or described, which can help improve the quality and usability of the ontology.
+
+6. Identifying Potentially Overspecified Class Definitions
+
+Title: Overspecified Class Detection
+Constraint Description: Identifies classes with potentially redundant or overlapping property definitions.
+Severity: Warning
+
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+SELECT ?class ?property1 ?property2 ?definition1 ?definition2
+WHERE {
+  ?class a owl:Class .
+  ?class ?property1 ?definition1 .
+  ?class ?property2 ?definition2 .
+  FILTER(?property1 != ?property2) 
+}
+
```

Comment:
I think a potential worry is that this query does not specify between rdfs:label and owl:equivalentClass. If the goal is to compare specific types of properties, you can add FILTER conditions to narrow down the results. You can add something like:

FILTER(STRSTARTS(STR(?property1), "http://www.w3.org/2000/01/rdf-schema#") || 
       STRSTARTS(STR(?property1), "http://www.w3.org/2004/02/skos/core#"))
FILTER(STRSTARTS(STR(?property2), "http://www.w3.org/2000/01/rdf-schema#") || 
       STRSTARTS(STR(?property2), "http://www.w3.org/2004/02/skos/core#"))

Reply:
Thank you!